### ISSUE TITLE
kubectl proxy appears to break WebSockets

### ISSUE THREAD

**Issue by errordeveloper** [weaveworks/scope#1427]
There is a a little unclear https://github.com/kubernetes/kubernetes/issues/24482, but we seem to be affected by that.

We need to confirm whether it’s the `kube proxy` that’s broken and we cannot do anything, or there is something we can do on our end to make it work.

**Comment by errordeveloper** [comment]
To reproduce, do this follow [K8s Anywhere setup](https://github.com/weaveworks/kubernetes-anywhere/blob/master/DOCKER_FOR_MAC.md) and deploy scope like this:
```
kubectl create -f https://gist.github.com/errordeveloper/9d9a995df05064ea3f600a0378de3aaf/raw/scope.json --validate=false
```

**Comment by errordeveloper** [comment]
xref kubernetes#25126

**Comment by errordeveloper** [comment]
@2opremio and I discussed this offline, and it looks like it’s not something we can fix soon, but I’d like to keep it open unless we would like to explicitly document the fact that Scope doesn't work with `kubectl proxy`.

**Comment by 2opremio** [comment]
I think there's little point on keeping this issue open when there's an upstream ticket already tracking it.

**Comment by errordeveloper** [comment]
That's true, I created this in the process of figuring it out. However, if
we close it now, we should consider that it's referenced in Kubernetes
repo, closing it would indicate as if it's no longer a problem for us,
while still really is a problem.

**Comment by satyamz** [comment]
Hi all, I am getting the same problem. I am not able to access the `weave-scope`  `live` mode (communicates using web sockets). However, I am able to access `pause` mode (Plain http). 
I am trying to access weave-scope using `/api/v1/namespaces/weave/services/weave-scope-app/proxy/` 

Is this issue resolved yet? If not, then please provide me some pointers so that I can try to work on it.

**Comment by errordeveloper** [comment]
@satyamz the issue has been fixed in Kubernetes 1.8 (see https://github.com/kubernetes/kubernetes/pull/49534). If you are using an older version of Kubernetes, please use `kubectl port-foward` (as describe in [the docs](https://www.weave.works/docs/scope/latest/installing/#k8s)).
